Bluetooth: GATT: Fixes

Vudentz opened the pull request:
Some small fixes to BT_GATT_CCC to actually the declare its storage internally and bt_gatt_indicate to allow indicating by UUID.

Fixes #18547 #18572

Vudentz committed:
Bluetooth: GATT: Add support for indicating by UUID

This adds support to provide an UUID to bt_gatt_indicate so API user
don't need to hardcode the attribute offset by hand.

Fixes #18572

Signed-off-by: Luiz Augusto von Dentz <luiz.von.dentz@intel.com>

Vudentz committed:
Bluetooth: GATT: Fix using variable size storage for CCC

This removes the necessity of registering the storage for CCC and make
it part of the declaration itself.

Fixes #18547

Signed-off-by: Luiz Augusto von Dentz <luiz.von.dentz@intel.com>

jhedberg commented:
Does this need backporting?

Vudentz commented:
> Does this need backporting?

Not really, there are in fact enhancements.

nvlsianpu commented:
cc @rakons

joerchan commented on subsys/bluetooth/host/gatt.c:
```diff
@@ -1634,18 +1638,37 @@ int bt_gatt_indicate(struct bt_conn *conn,
+	attr = params->attr;
+
+	handle = attr->handle ? : find_static_attr(attr);
 	if (!handle) {
```
It looks to me like the user still has to provide a valid `attr`? Or am I misunderstanding?

Vudentz replied:
Yes, it is used as start location so it can either point to the start of its services, more likely, or handle 0x0001.

Vudentz committed:
Bluetooth: GATT: Fix bt_gatt_notify_cb documentation

This changes to the sentence about how attribute parameter is used when
notifying by UUID to sound proper english.

Signed-off-by: Luiz Augusto von Dentz <luiz.von.dentz@intel.com>
